src/mscorlib/src/System/IO/UnmanagedMemoryStream.cs

```diff
@@ -334,10 +325,6 @@ public override long Position
                     throw new NotSupportedException(SR.NotSupported_UmsSafeBuffer);
                 if (!_isOpen) __Error.StreamIsClosed();
 
-                // Note: subtracting pointers returns an Int64.  Working around
```

Comment:
I think this should be:

```
long newPosition = (long)value - (long)_mem;
if (newPosition < 0)
    throw new ArgumentOutOfRangeException("offset", SR.ArgumentOutOfRange_UnmanagedMemStreamLength); 
if (value < _mem)
    throw new IOException(SR.IO_SeekBeforeBegin); 

Interlocked.Exchange(ref _position, newPosition);
```

to throw for the negative position case from my example; and to behave correctly for other similar cases.